hawkbit-security-integration/src/main/java/org/eclipse/hawkbit/security/ControllerPreAuthenticatedSecurityHeaderFilter.java

```diff
@@ -97,18 +101,22 @@ public HeaderAuthentication getPreAuthenticatedPrincipal(final TenantSecurityTok
     }
 
     @Override
-    public HeaderAuthentication getPreAuthenticatedCredentials(final TenantSecurityToken secruityToken) {
+    public Object getPreAuthenticatedCredentials(final TenantSecurityToken secruityToken) {
         final String authorityNameConfigurationValue = tenantAware.runAsTenant(secruityToken.getTenant(),
                 sslIssuerNameConfigTenantRunner);
         String controllerId = secruityToken.getControllerId();
         // in case of legacy download artifact, the controller ID is not in the
-        // URL path, so then
-        // we just use the common name header
+        // URL path, so then we just use the common name header
         if (controllerId == null || "anonymous".equals(controllerId)) {
             controllerId = secruityToken.getHeader(caCommonNameHeader);
         }
 
-        return new HeaderAuthentication(controllerId, authorityNameConfigurationValue);
+        List<String> knownHashes = splitMultiHash(authorityNameConfigurationValue);
```

Comment:
According to @kaizimmerm  pull-request https://github.com/eclipse/hawkbit/pull/294 about memory consumption you could initialize the HashSet with the `knownHashes` size.
`Set<HeaderAuthentication> multiHashes = Sets.newHashSetWithExpectedSize(knownHashes.size());`